.travis.yml

```diff
@@ -1,49 +1,67 @@
+language: scala
+scala: 2.12.4
+jdk: openjdk8
+
 cache:
   directories:
   - "~/.m2/repository"
   - "~/.cache/pip"
+  - $HOME/.ivy2/cache
+
+before_cache:
+  # Cleanup the cached directories to avoid unnecessary cache updates
+  - find $HOME/.ivy2/cache -name "ivydata-*.properties" -print -delete
+
+before_install:
+  - git fetch --tags
+
+stages:
+  - name: test
+  - name: release
+    if: ((branch = master AND type = push) OR (tag IS present)) AND NOT fork
+
 jobs:
   include:
   - stage: test
-    language: scala
-    scala: 2.12.4
-    jdk: openjdk8
-    install:
-    - mvn install -P !build-extras -DskipTests=true -Dmaven.javadoc.skip=true -B
     script:
-    - mvn test -P !build-extras -B
+    - sbt test
   - stage: test
-    language: python
-    python:
-    - 3.6
+    before_install:
+      - - wget http://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -O miniconda.sh
+        - chmod +x miniconda.sh
+        - ./miniconda.sh -b
+        - export PATH=/home/travis/miniconda3/bin:$PATH
+        - conda update --yes conda
+        - conda create --yes -n condaenv python=3.6
```

Comment:
Do we want to test more python versions? They're up to 3.9 now, and I don't have a good sense of what may work in 3.6 but not in newer versions.

Reply:
We might, but this is parity with what we had, so I'd do that in another PR.  I think we'll have to do it somewhat more manually since travis doesn't have multi-language support and I could only get the sbt steps working with language: scala.